src/prommis/uky/costing/models/tests/test_hd_ce.py

```diff
@@ -0,0 +1,807 @@
+            pyo.Param,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.engineering_and_drafting,
+            pyo.Param,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.number_of_units, pyo.Param
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint1,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint2,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint3,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint4,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint5,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint6,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint7,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint8,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint9,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.heat_loss_constraint10,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.operating_cost_eq,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.base_cost_per_unit_eq,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.capital_cost_eq,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.variable_operating_cost_per_unit_eq,
+            pyo.Constraint,
+        )
+        assert isinstance(
+            model.fs.hydrogen_decrepitation_furnace.costing.variable_operating_cost_eq,
+            pyo.Constraint,
+        )
+
+    @pytest.mark.component
+    def test_solve1(self, model):
+        results = solver.solve(model, tee=True)
+        assert_optimal_termination(results)
+
+    @pytest.mark.unit
+    def test_solved_model_diagnostics(self, model):
+        dt = DiagnosticsToolbox(model)
+        dt.assert_no_numerical_warnings()
+
+    @pytest.mark.component
```

Comment:
Probably need to update these values since they will be reported in `USD_Jan_2024` now.